raft/server.hh

```diff
@@ -43,8 +45,16 @@ public:
         bool enable_forwarding = true;
 
         // Max size of a single command, add_entry with a bigger command will throw command_is_too_big_error.
-        // The value of zero means no limit.
-        size_t max_command_size = 0;
+        // The following condition must be satisfied:
+        // max_command_memory_usage() <= max_log_size - max_snapshot_trailing_bytes
+        // A snapshot will be triggered if memory_usage reaches _config.max_log_size - _config.max_command_memory_usage(),
```

Comment:
Why is it triggered in this case?

Reply:
suppose we don't have this rule and consider this scenario:
max_log_size=10
max_command_memory_usage=7
* command of size 4 gets committed/applied, snapshot is not taken, as the limits are not reached
* command of 7 enters `add_entry_on_leader`. It will wait forever, as there is nothing that will trigger the snapshot.

This is weird condition, I agree, and that's why there is a follow-up (more complicated) patch, where `add_entry_on_leader` actively triggers snapshot in this situation.

Reply:
I understand the rationale you had in mind. My point is that this is an algorithm you chose, and it would be nice to explain it.

Reply:
This is not how the reader will understand the phrase "Why is it triggered in this case?" . I took it as just a question. Please state exactly what you mean. I'll adjust the comment.